src/main/java/com/iota/iri/service/validation/TransactionSolidifier.java

```diff
@@ -0,0 +1,342 @@
+package com.iota.iri.service.validation;
+
+import com.iota.iri.controllers.TransactionViewModel;
+import com.iota.iri.model.Hash;
+import com.iota.iri.network.pipeline.TransactionProcessingPipeline;
+import com.iota.iri.network.TransactionRequester;
+import com.iota.iri.service.snapshot.Snapshot;
+import com.iota.iri.service.snapshot.SnapshotProvider;
+import com.iota.iri.storage.Tangle;
+import com.iota.iri.utils.log.interval.IntervalLogger;
+import com.iota.iri.utils.thread.DedicatedScheduledExecutorService;
+import com.iota.iri.utils.thread.SilentScheduledExecutorService;
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+
+import static com.iota.iri.controllers.TransactionViewModel.PREFILLED_SLOT;
+import static com.iota.iri.controllers.TransactionViewModel.fromHash;
+
+/**
+ * A solidifier class for processing transactions. Transactions are checked for solidity, and missing transactions are
+ * subsequently requested. Once a transaction is solidified correctly it is placed into a broadcasting set to be sent to
+ * neighboring nodes.
+ */
+public class TransactionSolidifier {
+
+    private Tangle tangle;
+    private SnapshotProvider snapshotProvider;
+    private TransactionRequester transactionRequester;
+
+    /**
+     * Interval that the {@link #transactionSolidifierThread()} will scan at.
+     */
+    private static int RESCAN_INTERVAL = 500;
+
+    private static final IntervalLogger log = new IntervalLogger(TransactionSolidifier.class);
+
+    /**
+     * Executor service for running the {@link #transactionSolidifierThread()}.
+     */
+    private SilentScheduledExecutorService executorService = new DedicatedScheduledExecutorService(
+            "Transaction Solidifier", log.delegate());
+
+    /**
+     * A queue for processing transactions with the {@link #checkSolidity(Hash)} call. Once a transaction has been
+     * marked solid it will be placed into the {@link #transactionsToUpdate} queue.
+     */
+    private Deque<Hash> transactionsToSolidify = new ArrayDeque<>();
+    /**
+     * A queue for processing transactions with the {@link #updateSolidTransactions(Tangle, Snapshot, Hash)} call.
+     * Once the transaction is updated it is placed into the {@link #transactionsToBroadcast} set.
+     */
+    private Deque<Hash> transactionsToUpdate = new ArrayDeque<>();
+    /**
+     * A set of transactions that will be called by the {@link TransactionProcessingPipeline} to be broadcast to
+     * neighboring nodes.
+     */
+    private Set<TransactionViewModel> transactionsToBroadcast = new LinkedHashSet<>();
```

Comment:
Why not `ConcurrentHashMap.newKeySet()`?
See https://stackoverflow.com/questions/6720396/different-types-of-thread-safe-sets-in-java